### DISCUSSION TITLE
Profiling mmap files

### DISCUSSION THREAD

**Discussion by st4rl3ss** [darshan-hpc/darshan#746] [Q&A] [answered]
Hello there!
We are using darshan to profile a program that uses mmap. Unfortunately in the results all we get is a -1 for POSIX_MMAPS calls.

Should we activate some special flags to activate mmap profiling?

Thank you very much in advance.
Regards

**Comment by shanedsnyder** [comment, marked as answer]
Hi!

First of all, apologies for taking so long to get back to you. This got lost in my email and I'm not accustomed to checking our discussions channel like our issues/PRs.

As I recall, we disable our tracking of `mmap()` usage (we just mark the counters as `-1` as you see) inside our shared library due to some bad interactions with some other libraries we've seen on Cray systems in the past. We do still collect the info for statically-linked executables, but looks like your system is dynamically linking.

If it's any consolation, we only track the number of `mmap()` calls issued to each file. We do not have any way of tracking how much data is accessed via `mmap()`, offsets accessed, etc., so the data is of limited value, compared to what we capture with other interfaces. Since `mmap()` just provides pointers to memory applications can access directly, we are very limited in in what we can capture.
